### DISCUSSION TITLE
How do I queue playlists to play for a fixed amount of time without scheduling?

### DISCUSSION THREAD

**Discussion by pixelqiwi** [AzuraCast/AzuraCast#7408] [Q&A]
The title is self explanatory, but I'll describe what I want to do:

My station has 10+ (and counting) playlists that are differentiated by genres, and I've been trying to make it so each playlist plays for 30 minutes, after which a jingle plays and the next (randomly picked) playlist starts playing, rinse and repeat.

I've been told to use the schedule, but it would be a pain to edit the schedule each time a new playlist appears, plus I don't really like how fixed the schedule is. My thoughts were to use liquidsoap for that, but neither me, nor my friend who has an AzuraCast station, nor ChatGPT could come up with a way to make that work.

If you have any ideas or have done something like this, please teach me how you did it, I would greatly appreciate it.

**Comment by caturria** [comment]
Hi @pixelqiwi,
Right now this would be tricky to do. If you had a set amount of them (I.E. eight) and could time an entire rotation to fit entirely within a day, then you could build it using the existing schedule features (but you'd have to revamp the whole thing as soon as it changed).
I plan to propose and implement some more advanced scheduling features (conditional scheduling and clock wheels) that would make this easier, but I won't be able to start on anything for several weeks.
If you have some coding skills, I wouldn't suggest custom Liquidsoap for this, but rather I'd suggest writing an external script that rotates the playlists and generates schedules for them using the AzuraCast API (which BTW is excellent). This is how I currently manage scheduling almost 100 one-time-only playlists per day.

**Reply by pixelqiwi** [reply]
fair enough, will be looking forward to the day when there would be more schedule options. 
could you, like, give me a nudge in the right direction by sharing an example script or something? i'm anything but a coder/programmer so i'd take any help i can get, really. if not, i'm still thankful for your response! and still open to other ideas in case anyone happens to have any
